File: Easy/929. Unique Email Addresses/numUniqueEmails.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        unordered_set<string> s;
        for(string e : emails){
            string mail;
            for(char c : e){
                if(c == '.'){
                    continue;
                }
                if(c == '+' || c == '@'){
                    break;
                }
                mail.push_back(c);
            }
            mail += e.substr(e.find('@'));
            s.insert(mail);
        }
        return s.size();
    }
};

File: Easy/929. Unique Email Addresses/numUniqueEmails.java
class Solution {
    public int numUniqueEmails(String[] emails) {
        Set<String> s = new HashSet<>();
        for(String e : emails){
            String name = "";
            for(char c : e.toCharArray()){
                if(c == '.'){
                    continue;
                }
                if(c == '@' || c == '+'){
                    break;
                }
                name += c;
            }
            name += e.substring(e.indexOf('@'),e.length());
            s.add(name);
        }
         return s.size();
    }
}